### PR TITLE
Use Reppy instead of robotsparser

### PR DESCRIPTION
**Pull request by Parth-Vader**
Made the changes using https://github.com/scrapy/scrapy/pull/949 for https://github.com/scrapy/scrapy/issues/754.

### CODE REVIEW

File: scrapy/downloadermiddlewares/robotstxt.py [hunk 1]
```diff
@@ -83,21 +88,11 @@ def _logerror(self, failure, request, spider):
...
-                # If we found garbage, disregard it:,
-                # but keep the lookup cached (in self._parsers)
-                # Running rp.parse() will set rp state from
-                # 'disallow all' to 'allow any'.
-                pass
-        # stdlib's robotparser expects native 'str' ;
-        # with unicode input, non-ASCII encoded bytes decoding fails in Python2
+        policy = HeaderWithDefaultPolicy(default=1800, minimum=600)
+
+        rp=Robots.fetch(response.url, ttl_policy=policy)
```
Comment (redapple) [inline scrapy/downloadermiddlewares/robotstxt.py]: I understand that this performs an HTTP request to get the robots.txt file. Can you check?
We do not want to re-fetch the file, nor use a synchronous networking call (reppy uses python-requests as far as I understand)

File: requirements.txt [hunk 2]
```diff
@@ -8,3 +8,4 @@ six>=1.5.2
 PyDispatcher>=2.0.5
 service_identity
 parsel>=1.1
+reppy>=0.3.0
```
Comment (redapple) [inline requirements.txt]: Is 0.3.0 the first Python 3 compatible release?

**Comment by redapple** [comment]
Thank you @Parth-Vader for stepping in!

File: requirements.txt (hunk 2 above)
Comment (Parth-Vader) [inline requirements.txt]: Yes, I ran the tests using `py35`.

**Comment by redapple** [comment]
It looks like [either Cython or a C++ compiler](https://github.com/seomoz/reppy/blob/v0.4.6/setup.py#L39) is needed [to build reppy](https://github.com/seomoz/reppy/blob/v0.4.6/dev-requirements.txt) now.
Can you try updating Travis config to add Cython?
I'm not sure we want to add another non-python dependency to Scrapy by default. Do they have a non-C++/less-speedy/Python-only fallback?

**Comment by kmike** [comment]
@redapple relevant discussion: https://github.com/seomoz/reppy/pull/33

**Comment by Parth-Vader** [comment]
Version 0.3.3 seems to be Cython free : 
https://github.com/seomoz/reppy/blob/v0.3.3/dev-requirements.txt

**Comment by kmike** [comment]
I think depending on old and unsupported version of a library is not OK, and depending on a library which is hard to install is also not OK.

**Comment by Parth-Vader** [comment]
Agreed.
So, is Reppy the right option for the purpose?

**Comment by redapple** [comment]
It looks like reppy is not (anymore) the right option for Scrapy's needs.

**Comment by Gallaecio** [comment]
#3796 made it possible to configure which robots.txt parser to use, including a built-in adapter fopr Reppy.

**Pull request closed without merging**